### PR TITLE
Add io.kopia.KopiaUI

### PR DESCRIPTION
**Pull request by darkdragon-001**
Package [Kopia UI](https://kopia.io/) based on Debian package.

Inspired by:
- https://github.com/flathub/im.riot.Riot
- https://github.com/flathub/org.signal.Signal

Since this is a backup application it needs full access to host's file system.

### CODE REVIEW

File: io.kopia.KopiaUI.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                    "type": "script",
+                    "dest-filename": "kopia-ui.sh",
+                    "commands": [
+                        "export TMPDIR=\"$XDG_RUNTIME_DIR/app/$FLATPAK_ID\"",
+                        "exec zypak-wrapper /app/KopiaUI/kopia-ui \"$@\""
+                    ]
+                },
+                {
+                    "type": "file",
+                    "path": "io.kopia.KopiaUI.appdata.xml"
```
Comment (LeandroStanger) [inline io.kopia.KopiaUI.json:58]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

Comment (darkdragon-001) [reply io.kopia.KopiaUI.json:58]: It is not clear to me what exactly the difference is. Should I add a `<launchable>` attribute for the `.desktop` file?

File: io.kopia.KopiaUI.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
```
Comment (bilelmoussaoui) [inline io.kopia.KopiaUI.appdata.xml:2]:
```suggestion
<component type="desktop-application">
```

File: io.kopia.KopiaUI.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    <content_attribute id="language-humor">none</content_attribute>
+    <content_attribute id="language-discrimination">none</content_attribute>
+    <content_attribute id="social-chat">none</content_attribute>
+    <content_attribute id="social-info">none</content_attribute>
+    <content_attribute id="social-audio">none</content_attribute>
+    <content_attribute id="social-location">none</content_attribute>
+    <content_attribute id="social-contacts">none</content_attribute>
+    <content_attribute id="money-purchasing">none</content_attribute>
+    <content_attribute id="money-gambling">none</content_attribute>
+  </content_rating>
```
Comment (bilelmoussaoui) [inline io.kopia.KopiaUI.appdata.xml:52]:
```suggestion
  <content_rating type="oars-1.1" />
```

File: io.kopia.KopiaUI.json [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    "finish-args": [
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=pulseaudio",
+        "--share=network",
+        "--device=dri",
+        "--filesystem=host",
+        "--talk-name=org.freedesktop.Notifications",
+        "--talk-name=org.kde.StatusNotifierWatcher",
+        "--own-name=org.kde.StatusNotifierItem-2-1"
```
Comment (bilelmoussaoui) [inline io.kopia.KopiaUI.json:21]: This won't work.

File: io.kopia.KopiaUI.json [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    "separate-locales": false,
+    "rename-desktop-file": "kopia-ui.desktop",
+    "rename-icon": "kopia-ui",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=pulseaudio",
+        "--share=network",
+        "--device=dri",
+        "--filesystem=host",
```
Comment (bilelmoussaoui) [inline io.kopia.KopiaUI.json:18]: What does it need host access for?

File: io.kopia.KopiaUI.json (hunk 4 above)
Comment (darkdragon-001) [reply io.kopia.KopiaUI.json:21]: What is _this_? I am using the flatpak myself for several days without noticing any problems...

File: io.kopia.KopiaUI.json (hunk 5 above)
Comment (darkdragon-001) [reply io.kopia.KopiaUI.json:18]: It is a backup program. In order to backup the whole system, it needs access to the whole system.

File: io.kopia.KopiaUI.json (hunk 4 above)
Comment (bilelmoussaoui) [reply io.kopia.KopiaUI.json:21]: It's used for tray icons. You should use `--own-name=org.kde.*`

File: io.kopia.KopiaUI.appdata.xml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  <project_license>Apache-2.0</project_license>
+  <developer_name>Kopia Authors</developer_name>
+  <summary>Backup solution</summary>
+  <metadata_license>CC0-1.0</metadata_license>
+  <url type="homepage">https://kopia.io/</url>
+  <url type="bugtracker">https://github.com/flathub/io.kopia.KopiaUI/issues</url>
+  <description>
+    <p>
+      Cross-platform backup tool for Windows, macOS &amp; Linux with fast, incremental backups, client-side end-to-end encryption, compression and data deduplication. CLI and GUI included.
+    </p>
```
Comment (bilelmoussaoui) [inline io.kopia.KopiaUI.appdata.xml:14]: Please add a note here that this package is not supported by upstream, see https://flathub.org/apps/details/com.amazon.Workspaces for an example

File: io.kopia.KopiaUI.json [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        {
+            "name": "kopia-ui",
+            "buildsystem": "simple",
+            "build-commands": [
+                "ar x kopia-ui_*.deb",
+                "rm -f kopia-ui_*.deb",
+                "tar xf data.tar.xz",
+                "rm -f control.tar.gz data.tar.xz debian-binary",
+                "cp -r usr/* opt/* /app",
+                "rm -rf /app/share/icons/hicolor/1024x1024/",
```
Comment (bilelmoussaoui) [inline io.kopia.KopiaUI.json:33]: Use cleanup instead.

**Comment by bilelmoussaoui** [comment]
Have you reached out to the upstream developers to see if they were interested in (co)-maintain their app on Flathub?

File: io.kopia.KopiaUI.json (hunk 4 above)
Comment (darkdragon-001) [reply io.kopia.KopiaUI.json:21]: Tray icons are working for me. `org.kde.*` looks too permissive for my taste. See https://github.com/flathub/org.keepassxc.KeePassXC/pull/61/commits/aef2c6ea0a9f455eb17c1f19038e017c535f00f7 for a detailed explanation.

**Comment by darkdragon-001** [comment]
> Have you reached out to the upstream developers to see if they were interested in (co)-maintain their app on Flathub?

See kopia/kopia#475. I was talking a bit more on Slack and they need some technical help but are in general interested in an official release on Flathub. So I think they should have maintainer rights at least!

File: io.kopia.KopiaUI.json (hunk 7 above)
Comment (darkdragon-001) [reply io.kopia.KopiaUI.json]: Like this?

**Comment by nedrichards** [comment]
bot, build io.kopia.KopiaUI

**Pull request closed without merging**